php-classes/Slate/CBL/StudentCompetency.php

```diff
@@ -53,16 +64,349 @@ class StudentCompetency extends \ActiveRecord
         ]
     ];
 
-    public static function isCurrentLevelComplete($Student, $Competency)
+    public static $dynamicFields = [
+        'completion' => [
+            'getter' => 'getCompletion'
+        ],
+        'demonstrationsLogged' => [
+            'getter' => 'getDemonstrationsLogged'
+        ],
+        'demonstrationsComplete' => [
+            'getter' => 'getDemonstrationsComplete'
+        ],
+        'demonstrationsAverage' => [
+            'getter' => 'getDemonstrationsAverage'
+        ],
+        'demonstrationData' => [
+            'getter' => 'getDemonstrationData'
+        ],
+        'effectiveDemonstrationsData' => [
+            'getter' => 'getEffectiveDemonstrationsData'
+        ]
+    ];
+
+    public function getCompletion()
+    {
+        return [
+            'StudentID' => $this->StudentID,
+            'CompetencyID' => $this->CompetencyID,
+            'currentLevel' => $this->Level,
+            'baselineRating' => $this->BaselineRating,
+            'demonstrationsLogged' => $this->getDemonstrationsLogged(),
+            'demonstrationsComplete' => $this->getDemonstrationsComplete(),
+            'demonstrationsAverage' => $this->getDemonstrationsAverage()
+        ];
+    }
+
+    public function calculateStartingRating()
+    {
+        $validSkills = DB::allRecords(
```

Comment:
@nbey let's use `DB::table('ID', '...', [...])` and then `$validSkillIds = array_keys($validSkills)`